src/lua/luafile.cc

```diff
@@ -73,6 +74,22 @@ LuaFile* subFile(LuaFile* wrapper, uint64_t start, int64_t size) {
 }
 LuaFile* uvFifo(const char* address, int port) { return new LuaFile(new PCSX::UvFifo(address, port)); }
 
+LuaServer* uvFifoListener() {return new LuaServer(new PCSX::UvFifoListener());}
+
+void startListener(LuaServer* server, unsigned port, void (*cb)(LuaFile* fifo)) { server->m_listener->start(port, PCSX::g_system->getLoop(), &server->m_async, [cb, server](PCSX::UvFifo* fifo) {
+        if (fifo) {
+            cb(new LuaFile(fifo));
+        } else {
+            server->m_async.data = server;
+            uv_close(reinterpret_cast<uv_handle_t*>(&server->m_async), [](uv_handle_t* handle) {});
+        }
+    });
+}
+
+void stopListener(LuaServer* server) {
+   server->m_listener->stop();
```

Comment:
Probably need a status variable to avoid double stops. It needs to be "STARTED", "STOPPING", "STOPPED".